tmt/cli.py

```diff
@@ -687,6 +687,28 @@ def create(context, name, template, force, **kwargs):
     tmt.Plan.create(name, template, context.obj.tree.root, force)
 
 
+@plans.command()
+@click.pass_context
+@name_filter_condition_long
+@click.option(
+    '--format', 'format_', default='yaml', show_default=True, metavar='FORMAT',
+    help='Output format.')
+@click.option(
+    '-d', '--debug', is_flag=True,
+    help='Provide as much debugging details as possible.')
+def export(context, format_, **kwargs):
+    """
+    Export plans into desired format.
+
+    Regular expression can be used to filter plans by name.
+    Use '.' to select plans under the current working directory.
+    """
+    tmt.Plan._save_context(context)
+
+    for plan in context.obj.tree.plans():
+        echo(plan.export(format_=format_))
```

Comment:
The output should be a list of dictionaries. Currently multiple exported plans are separated with a blank space which is not possible to easily process by a `yaml` parser.

Reply:
@lukaszachy, just to confirm: The expected output format is a list of dictionaries, not a dictionary of dictionaries, right?

Reply:
That is good question. Will 'name' be part of final dictionary? We shouldn't loose this information. Currently there is no plan 's name in the exported data so dictionary of dictionaries might be better (key = name of plan, value = plan data)

@thrix  what is easier / makes more sense to parse via yq/jq?

Reply:
I think that list of dicts is a bit better because it explicitly defines the order. Which might be (sometimes) necessary/useful. Yes, `name` has to be included as well (already commented around line 955).

Reply:
@thrix, @happz, are you ok with the list of dictionaries.

Reply:
Can you share an example of the output?

Reply:
It could look like this:

```yaml
  - name: /plan/plan1
    discover:
        how: fmf
        name: default
    execute:
        how: tmt
        name: default
    summary: First plan
    enabled: true
    order: 50
    link: []

  - name: /plan/plan1
    discover:
        how: fmf
        name: default
    execute:
        how: tmt
        name: default
    summary: Second plan
    enabled: true
    order: 50
    link: []
```

Reply:
Looks good to me. I agree `name` shouldn't be lost, and I also like the possibility of safe ordering - it's probably of no use now, but one day, `tmt` may gain an option to enforce some kind of ordering of plans/tests, and then a list would be needed. Even when an order isn't reliable nor guaranteed, I'd prefer a list.

Reply:
LGTM

Reply:
Hello, the output looks like this:
1. plan name
2. inherited Core keys
3. extra_L2_keys `['context', 'environment', 'gate']` if available
4. `tmt` steps `['discover', 'provision', 'prepare', 'execute', 'report', 'finish']` if available

```
  - name: /plan/plan_L2_context
    summary: Testing plans to export
    description: Testing data for test the feature to export plans to yaml or dict
    enabled: true
    order: 50
    link: []
    context:
        component: dash
    discover:
        how: fmf
        name: default
    execute:
        how: tmt
        name: default
  - name: /plan/plan_L2_environment
    summary: Testing plans to export
    description: Testing data for test the feature to export plans to yaml or dict
    enabled: true
    order: 50
    link: []
    environment:
        RELEASE: f35
    discover:
        how: fmf
        name: default
    execute:
        how: tmt
        name: default
  - name: /plan/plan_L2_gate
    summary: Testing plans to export
    description: Testing data for test the feature to export plans to yaml or dict
    enabled: true
    order: 50
    link: []
    gate:
      - merge-pull-request
      - add-build-to-update
      - add-build-to-compose
    discover:
        how: fmf
        name: default
    execute:
        how: tmt
        name: default
  - name: /plan/plan_no_L2_key
    summary: Testing plans to export
    description: Testing data for test the feature to export plans to yaml or dict
    enabled: true
    order: 50
    link: []
    discover:
        how: fmf
        name: default
    execute:
        how: tmt
        name: default
```